hydra-node/test/Hydra/Chain/Direct/Contract/Close.hs

```diff
@@ -170,13 +199,36 @@ genCloseMutation (tx, _utxo) =
   mutateCloseUTxOHash :: Gen (TxOut CtxTx)
   mutateCloseUTxOHash = do
     mutatedUTxOHash <- genHash
-    pure $ changeHeadOutputDatum (mutateState mutatedUTxOHash) headTxOut
+    pure $ changeHeadOutputDatum (mutateHash mutatedUTxOHash) headTxOut
 
-  mutateState mutatedUTxOHash = \case
-    Head.Closed{snapshotNumber, parties} ->
+  mutateHash mutatedUTxOHash = \case
+    Head.Closed{snapshotNumber, parties, contestationDeadline} ->
       Head.Closed
         { snapshotNumber
         , utxoHash = toBuiltin mutatedUTxOHash
         , parties
+        , contestationDeadline
+        }
+    st -> error $ "unexpected state " <> show st
+
+  mutateClosedContestationDeadline :: Gen (TxOut CtxTx)
+  mutateClosedContestationDeadline = do
+    -- NOTE: we need to be sure the generated contestation period is large enough to have an impact on the on-chain
+    -- deadline computation, which means having a resolution of seconds instead of the default picoseconds
+    contestationPeriodSeconds <- arbitrary @Integer `suchThat` (/= healthyContestationPeriodSeconds)
+    pure $ changeHeadOutputDatum (mutateContestationDeadline contestationPeriodSeconds) headTxOut
+
+  mutateContestationDeadline contestationPeriod = \case
+    Head.Closed{snapshotNumber, utxoHash, parties} ->
+      Head.Closed
+        { snapshotNumber
+        , utxoHash
+        , parties
+        , contestationDeadline =
+            -- FIXME: we don't have useful functions for manipulating time on-chain seemingly
```

Comment:
What exactly is it to fix here? Sounds more like a complaint but `slotNoToPOSIXTime` works in this case?